src/quill-editor.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, InjectionToken, Inject, PLATFORM_ID } from "@angular/core";
+import { QuillConfig } from "./quill-editor.interfaces";
+import { dynamicImportCommonJs } from "./helpers";
+import { isPlatformServer } from "@angular/common";
+
+export const QUILL_CONFIG_TOKEN = new InjectionToken<QuillConfig>("config");
+
+import { Quill } from "quill";
+import { defaultModules } from "./quill-defaults";
+let QuillNamespace: any;
+const QuillPromise = dynamicImportCommonJs<Quill | typeof Quill>(() =>
+  import(/* webpackChunkName: "quill" */ "quill").then(m => m.Quill)
+);
+
+@Injectable({ providedIn: "root" })
+export class QuillEditorService {
+  constructor(
+    @Inject(QUILL_CONFIG_TOKEN) private config?: QuillConfig,
+    @Inject(PLATFORM_ID) private platformId: any
+  ) {
+    this.init();
+  }
+  async init() {
```

Comment:
a constructor should not do async stuff. because instantiation should be synchron. In that case you could use the service but initialisation is not ready.

Better would be to add something like a "getConfig" method that does it explicitly (but only once, e.g. with  private property if the registration of modules has run

Reply:
Well in both the cases the init will remain to be async, as it will dynamically import the quill. 

If we will not use it in the constructor, then we will have to create additional code to check if the registration is already done or not yet, if not then only register the modules.

Reply:
that is what i mean. just a little additional logic and it should be fine.